lib/Doctrine/ORM/Utility/IdentifierFlattener.php

```diff
@@ -85,6 +85,9 @@ public function flattenIdentifier(ClassMetadata $class, array $id)
                 }
 
                 $flatId[$field] = implode(' ', $associatedId);
+            } elseif (array_key_exists($field, $id)) {
```

Comment:
Can `$id[$field]` be `null`? If that isn't the case, `isset()` is able to deliver better performance.